modules/programs/radio-active.nix

```diff
@@ -0,0 +1,194 @@
+              programs.radio-active.settings.AppConfig = {
+                loglevel = "debug";
+                limit = 41;
+                sort = "votes";
+                filter = "none";
+                volume = 68;
+                filepath = "/home/{user}/recordings/radioactive/";
+                filetype = "mp3";
+                player = "ffplay";
+              };
+              ```
+
+              ## Output
+
+              ```
+              [AppConfig]
+              loglevel = debug
+              limit = 41
+              sort = votes
+              filter = none
+              volume = 68
+              filepath = /home/{user}/recordings/radioactive/
+              filetype = mp3
+              player = ffplay
+              ```
+            '';
+
+            type = attrsOf (oneOf [
+              nonEmptyStr
+              nonnegative
+            ]);
+          };
+
+          aliases = mkOption {
+            description = ''
+              Key/value pairs where the key is name of radio station and value is URL
+            '';
+
+            example = ''
+              ## Input
+
+              ```nix
+              programs.radio-active.aliases = {
+                "Deep House Lounge" = "http://198.15.94.34:8006/stream";
+              };
+              ```
+
+              ## Output
+
+              ```
+              Deep House Lounge==http://198.15.94.34:8006/stream
+              ```
+            '';
+
+            default = { };
+            type = attrsOf nonEmptyStr;
+          };
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    ## TODO: test that dependency `postPatch` modifications works at runtime
+    home.packages =
+      let
+        player = attrByPath [ "settings" "AppConfig" "player" ] "ffplay" cfg;
+
+        radio-active =
+          if player == "ffplay" then
+            pkgs.radio-active
+          else if player == "mpv" then
+            pkgs.radio-active.overrideAttrs (
+              finalAttrs: previousAttrs: {
+                postPatch = ''
+                  ${previousAttrs.postPatch}
+
+                  substituteInPlace radioactive/mpv.py \
+                    --replace-fail 'self.exe_path = which(self.program_name)' \
+                    'self.exe_path = "${lib.getExe pkgs.mpv}"'
+                '';
+              }
+            )
+          else if player == "vlc" then
+            pkgs.radio-active.overrideAttrs (
+              finalAttrs: previousAttrs: {
+                postPatch = ''
+                  ${previousAttrs.postPatch}
+
+                  substituteInPlace radioactive/vlc.py \
+                    --replace-fail 'self.exe_path = which(self.program_name)' \
+                    'self.exe_path = "${lib.getExe pkgs.vlc}"'
+                '';
+              }
+            )
+          else
+            lib.warn ''
+              Unknown player defined in `config.programs.radio-active.AppConfig.player`
+            '' pkgs.radio-active;
```

Comment:
Warnings should go in the `warnings` option, that way they are presented consistently. See, e.g., https://github.com/nix-community/home-manager/blob/173a29f735c69950cfeaac310d7e567115976be0/modules/programs/bun.nix#L51-L55